src/ManifestStrategy/GlossaryManifestStrategy.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\ManifestStrategy;
+
+use SprykerSdk\Integrator\Dependency\Console\InputOutputInterface;
+
+class GlossaryManifestStrategy extends AbstractManifestStrategy
+{
+    /**
+     * @var int
+     */
+    protected const GLOSSARY_LINE_PARTS_COUNT = 3;
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return 'glossary-key';
+    }
+
+    /**
+     * @param array<string, array<string, string>> $manifest
+     * @param string $moduleName
+     * @param \SprykerSdk\Integrator\Dependency\Console\InputOutputInterface $inputOutput
+     * @param bool $isDry
+     *
+     * @return bool
+     */
+    public function apply(array $manifest, string $moduleName, InputOutputInterface $inputOutput, bool $isDry): bool
+    {
+        $glossaryFilePath = $this->config->getGlossaryFilePath();
+        if (!file_exists($glossaryFilePath)) {
+            $inputOutput->writeln(sprintf(
+                'File `%s` does not exist. Please check file path or customize using `IntegratorConfig::getGlossaryFilePath()`.',
+                $glossaryFilePath,
+            ), InputOutputInterface::DEBUG);
+
+            return false;
+        }
+
+        $existingGlossaryFileLines = $this->getGlossaryExistingFileLines();
+        $updatedGlossaryFileLines = $this->getUpdatedGlossaryFileLines(
+            $manifest,
+            $existingGlossaryFileLines,
+        );
+        $resultGlossaryFileLines = array_merge($existingGlossaryFileLines, $updatedGlossaryFileLines);
+        if (!$isDry) {
+            file_put_contents($this->config->getGlossaryFilePath(), implode("\n", $resultGlossaryFileLines));
+        }
+        $inputOutput->writeln(sprintf(
+            'Glossary file `%s` successfully updated',
+            $glossaryFilePath,
+        ), InputOutputInterface::DEBUG);
+
+        return true;
+    }
+
+    /**
+     * @param array<string, array<string, string>> $manifest
+     * @param array<array-key, string> $existingGlossaryFileLines
+     *
+     * @return array<array-key, string>
+     */
+    protected function getUpdatedGlossaryFileLines(
+        array $manifest,
+        array $existingGlossaryFileLines
+    ): array {
+        $mappedGlossaryLinesByGlossaryKeysAndLanguages = $this->createMappedGlossaryLinesByGlossaryKeysAndLanguages(
+            $existingGlossaryFileLines,
+        );
+        $updatingGlossaryFileLines = [];
+        foreach ($manifest as $manifestKey => $manifestValue) {
```

Comment:
Just to be clear, manifest key is a glossary key? If so, please rename it accordingly. In my understanding `manifest key` is a type of manifest. E.g. `glossary-key`. So it would be nice to rename this in order to make code more readable